File: quickstart_guides/arrays_strings/python/sort_string.py
"""
Title: Sort string

Problem:
    Given a string of lowercase characters from ‘a’ – ‘z’. We need to write a
    program to print the characters of this string in sorted order.

Execution: python sort_string.py
"""
import unittest


def sort_string(s: str) -> str:
    return "".join(sorted(s))


class TestSortString(unittest.TestCase):
    """Unit test for sort_string."""

    def test_1(self):
        s = "bbccdefbbaa"
        self.assertEqual(sort_string(s), "aabbbbccdef")


if __name__ == "__main__":
    unittest.main()

File: quickstart_guides/arrays_strings/java/SortString.java
/*
 *   Title: Sort string
 *
 *   Problem:
 *   Given a string of lowercase characters from ‘a’ – ‘z’. We need to write a
 *   program to print the characters of this string in sorted order.
 *
 *   Execution: javac SortString.java && java SortString
 */

import java.util.Arrays;

public class SortString {
    public static String sortString(String s) {
        char[] arr = s.toCharArray();
        Arrays.sort(arr);
        return String.valueOf(arr);
    }

    public static void main(String[] args) {
        assert sortString("bbccdefbbaa") == "aabbbbccdef";
        assert sortString("cba") == "abc";
        System.out.println("Passed all test cases");
    }
}
